packages/kbn-test/src/kbn/kbn_test_config.ts

```diff
@@ -16,26 +16,34 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
-import { kibanaTestUser } from './users';
 import url from 'url';
+import { kibanaTestUser } from './users';
+
+interface UrlParts {
+  protocol: string;
+  hostname: string;
+  port: number;
+  auth: string;
+  username: string;
+  password: string;
+}
 
 export const kbnTestConfig = new (class KbnTestConfig {
   getPort() {
     return this.getUrlParts().port;
   }
 
-  getUrlParts() {
+  getUrlParts(): UrlParts {
     // allow setting one complete TEST_KIBANA_URL for ES like https://elastic:changeme@example.com:9200
     if (process.env.TEST_KIBANA_URL) {
       const testKibanaUrl = url.parse(process.env.TEST_KIBANA_URL);
       return {
-        protocol: testKibanaUrl.protocol.slice(0, -1),
-        hostname: testKibanaUrl.hostname,
-        port: parseInt(testKibanaUrl.port, 10),
-        auth: testKibanaUrl.auth,
-        username: testKibanaUrl.auth.split(':')[0],
-        password: testKibanaUrl.auth.split(':')[1],
+        protocol: testKibanaUrl.protocol!.slice(0, -1),
```

Comment:
I think we should validate `testKibanaUrl.protocol` and friends before asserting that they're non-null

Reply:
we could. I kept the existing API

Reply:
Hmm, were the non-null assertions necessary then? In my eyes non-null assertions are like using `any`, it usually shouldn't be necessary but now I see that this is basically the same code but with a bunch of non-null assertions..

Reply:
Hmm, were the non-null assertions necessary then? In my eyes non-null assertions are like using `any`, it usually shouldn't be necessary but now I see that this is basically the same code but with a bunch of non-null assertions..